catroid/src/main/java/org/catrobat/catroid/ui/BaseActivity.java

```diff
@@ -137,4 +149,23 @@ public void addToRequiresPermissionTaskList(RequiresPermissionTask task) {
 	public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
 		permissionRequestActivityExtension.onRequestPermissionsResult(this, requestCode, permissions, grantResults);
 	}
+
+	public static void removeExifData() {
+		File oldFile = new File(CAMERA_CACHE_DIR, TMP_IMAGE_FILE_NAME + ".jpg");
+		File file = new File(CAMERA_CACHE_DIR, TMP_IMAGE_FILE_NAME + "___temp.jpg");
+		try (FileInputStream inputStream = new FileInputStream(oldFile);
+				FileOutputStream outputStream = new FileOutputStream(file)) {
+
+			new ExifRewriter().removeExifMetadata(inputStream, outputStream);
+
+			if (!oldFile.delete()) {
+				Log.e(TAG, "removeExifData: Failed to delete old file");
+			}
+			if (!file.renameTo(new File(CAMERA_CACHE_DIR, TMP_IMAGE_FILE_NAME + ".jpg"))) {
```

Comment:
You can simply use `file.renameTo(oldFile)` here as you already defined this file.